Build fails on 32 bit architectures

Hi 
Trying build for RPI GOOS=linux GOARCH=arm CGO_ENABLED=1
get build failure
```
# github.com/tinyzimmer/go-glib/glib
vendor/github.com/tinyzimmer/go-glib/glib/gobject.go:247:22: type [1073741824]int larger than address space
vendor/github.com/tinyzimmer/go-glib/glib/gobject.go:247:22: type [1073741824]int too large
vendor/github.com/tinyzimmer/go-glib/glib/gobjectclass.go:49:25: type [1073741824]*_Ctype_struct__GParamSpec larger than address space
vendor/github.com/tinyzimmer/go-glib/glib/gobjectclass.go:49:25: type [1073741824]*_Ctype_struct__GParamSpec too large
vendor/github.com/tinyzimmer/go-glib/glib/gparamspec.go:74:27: type [1073741824]_Ctype_struct__GFlagsValue larger than address space
vendor/github.com/tinyzimmer/go-glib/glib/gparamspec.go:74:27: type [1073741824]_Ctype_struct__GFlagsValue too large

```

--- a/glib/gobject.go
+++ b/glib/gobject.go
@@ -9,6 +9,7 @@ import "C"
 import (
 	"errors"
 	"fmt"
+	"math"
 	"runtime"
 	"unsafe"
 )
@@ -244,7 +245,8 @@ func (v *Object) ListInterfaces() []string {
 	}
 	defer C.g_free((C.gpointer)(ifaces))
 	out := make([]string, int(size))
-	for _, t := range (*[1 << 30]int)(unsafe.Pointer(ifaces))[:size:size] {
+
+	for _, t := range (*[(math.MaxInt32 - 1) / unsafe.Sizeof(int(0))]int)(unsafe.Pointer(ifaces))[:size:size] {
 		out = append(out, Type(t).Name())
 	}
 	return out

--- a/glib/gobjectclass.go
+++ b/glib/gobjectclass.go
@@ -7,7 +7,10 @@ GObjectClass *  toGObjectClass  (void *p)  { return (G_OBJECT_CLASS(p)); }
 */
 import "C"
 
-import "unsafe"
+import (
+	"math"
+	"unsafe"
+)
 
 // ObjectClass is a binding around the glib GObjectClass. It exposes methods
 // to be used during the construction of objects backed by the go runtime.
@@ -46,7 +49,8 @@ func (o *ObjectClass) ListProperties() []*ParamSpec {
 	}
 	defer C.g_free((C.gpointer)(props))
 	out := make([]*ParamSpec, 0)
-	for _, prop := range (*[1 << 30]*C.GParamSpec)(unsafe.Pointer(props))[:size:size] {
+
+	for _, prop := range (*[(math.MaxInt32 - 1) / unsafe.Sizeof((*C.GParamSpec)(nil))]*C.GParamSpec)(unsafe.Pointer(props))[:size:size] {
 		out = append(out, ToParamSpec(unsafe.Pointer(prop)))
 	}
 	return out

--- a/glib/gparamspec.go
+++ b/glib/gparamspec.go
@@ -18,6 +18,7 @@ GFlagsValue *   getParamSpecFlags     (GParamSpec * p, guint * size)
 import "C"
 
 import (
+	"math"
 	"unsafe"
 )
 
@@ -71,7 +72,8 @@ func (p *ParamSpec) GetFlagValues() []*FlagsValue {
 	gFlags := C.getParamSpecFlags(p.paramSpec, &gSize)
 	size := int(gSize)
 	out := make([]*FlagsValue, size)
-	for idx, flag := range (*[1 << 30]C.GFlagsValue)(unsafe.Pointer(gFlags))[:size:size] {
+
+	for idx, flag := range (*[(math.MaxInt32 - 1) / unsafe.Sizeof(C.GFlagsValue{})]C.GFlagsValue)(unsafe.Pointer(gFlags))[:size:size] {
 		out[idx] = &FlagsValue{
 			Value:     int(flag.value),
 			ValueNick: C.GoString(flag.value_nick),

